### PR TITLE
Remove dead code from tools repo

### PR DESCRIPTION
**Pull request by mozbhearsum**
This removes what I think is all the dead code (and no undead code!) from the tools repo. The goal here is to make it easier to see what actually matters, so we know what we need to relocate before getting rid of the repo. Most of what was removed was directly buildbot related (eg: manage-masters.py) or indirectly buildbot related (eg: hgtool.py). The things that remain are:
* Balrog submission library
* Update verify
* Final verify
* Signing server scripts + libs
* MINIDUMP_STACKWALK binaries
* gittool.py

This patch _is_ removing some things currently referenced by in-tree code, but I think that code is also dead. Eg: the antivirus mozharness script uses extract_and_run_command.py, but we don't run it anymore. Periodic file updates talk about the tools repo, but don't use it (we imported that code into the tree). Merge day scripts reference tools repo but don't use anything from it in any way that I can find.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +0,0 @@
-# Tools [![Build Status](https://travis-ci.org/mozilla/build-tools.png)](https://travis-ci.org/mozilla/build-tools)
```
Comment (tomprince) [inline README.md]: We should leave some sort of README here.

File: buildfarm/maintenance/update_merge_day_wiki.sh [hunk 2]
```diff
@@ -1,143 +0,0 @@
-#!/bin/bash -e
```
Comment (tomprince) [inline buildfarm/maintenance/update_merge_day_wiki.sh]: This file is still referenced in the mergeday docs. (Although I think it is broken)

File: buildfarm/maintenance/wiki_functions.sh [hunk 3]
```diff
@@ -1,105 +0,0 @@
-#!/bin/bash
```
Comment (tomprince) [inline buildfarm/maintenance/wiki_functions.sh]: And this is used by the wiki script above.

File: buildfarm/utils/gittool.py [hunk 4]
```diff
@@ -1,81 +0,0 @@
-#!/usr/bin/python
```
Comment (tomprince) [inline buildfarm/utils/gittool.py]: The PR comment said this wasn't deleted.

Comment (mozbhearsum) [reply buildfarm/utils/gittool.py]: Whoops, forgot to push the commit that restored it!

File: lib/python/kickoff/__init__.py [hunk 5]
```diff
@@ -1,381 +0,0 @@
-import re
```
Comment (tomprince) [inline lib/python/kickoff/__init__.py]: Fairly certain this package is used by releaserunner3.

File: lib/python/build/paths.py [hunk 6]
```diff
@@ -1,37 +0,0 @@
-import os
```
Comment (tomprince) [inline lib/python/build/paths.py]: Stuff in here (and maybe other stuff in this package) appears to be used in the balrog code.

File: lib/python/release/config.py [hunk 7]
```diff
@@ -1,17 +0,0 @@
-import logging
```
Comment (tomprince) [inline lib/python/release/config.py]: Code in this package also appears to be used in balrog.

File: lib/python/build/paths.py (hunk 6 above)
Comment (mozbhearsum) [reply lib/python/build/paths.py]: Which Balrog code? It's not referenced anywhere in lib/python/balrog. That code does use release.paths, though, which still exists.

File: lib/python/util/git.py [hunk 8]
```diff
@@ -1,414 +0,0 @@
-"""Functions for interacting with hg"""
```
Comment (tomprince) [inline lib/python/util/git.py]: Used by gittool and uses some other bits of this package.

File: lib/python/util/sendmail.py [hunk 9]
```diff
@@ -1,17 +0,0 @@
-import email.utils
```
Comment (tomprince) [inline lib/python/util/sendmail.py]: Used by kickoff.

File: lib/python/util/hg.py [hunk 10]
```diff
@@ -1,669 +0,0 @@
-"""Functions for interacting with hg"""
```
Comment (tomprince) [inline lib/python/util/hg.py]: Used by chuncked-verfiy.py

File: release/README.txt [hunk 11]
```diff
@@ -1,63 +0,0 @@
-Mozilla Build Verification Scripts
```
Comment (tomprince) [inline release/README.txt]: Seems to have some relevant documentation?

**Comment by MihaiTabara** [comment]
We still use in https://github.com/mozilla-releng/releasewarrior-2.0/blob/master/docs/mergeduty/howto.md#update-wiki-versions the two wiki-scripts

https://hg.mozilla.org/build/tools/raw-file/default/buildfarm/maintenance/wiki_functions.sh
https://hg.mozilla.org/build/tools/raw-file/default/buildfarm/maintenance/update_merge_day_wiki.sh

Also, hgext/robustcheckout.py may be used in indirectly in mergeduty AFAIK (https://hg.mozilla.org/mozilla-central/file/tip/testing/mozharness/scripts/merge_day/gecko_migration.py#l446)

This one is still sending the "Build of emails" for releaserunner3.
I don't think we should touch lib/python/ just yet, not until releaserunner3 is not defunct. There's lots of stuff used there. Also some stuff we haven't ported yet, like bug creation.

File: lib/python/build/paths.py (hunk 6 above)
Comment (tomprince) [reply lib/python/build/paths.py]: You appear to be correct, and I missed that `release.versions` was changed not deleted.

**Comment by mozbhearsum** [comment]
>  I don't think we should touch lib/python/ just yet, not until releaserunner3 is not defunct. There's lots of stuff used there. Also some stuff we haven't ported yet, like bug creation.

I don't particularly want to keep stuff around because we think we might be using it -- that's going to make it much more difficult to kill this repo off. It's nearly impossible to prove that module X is not used anywhere (given than we clone and use tools from so many different places), so I think we'd be better off deleting everything that we can't find a usage for, and re-add anything that causes bustage when removed.

Clearly I need to widen my search for usage though!

File: lib/python/vendor/python-dateutil-2.6.0/ci_tools/retry.bat [hunk 12]
```diff
@@ -1,18 +1,18 @@
...
-set n_retries=5
-
-for /l %%x in (1, 1, %n_retries%) do (
-  call %*
-  if not ERRORLEVEL 1 EXIT /B 0
-  timeout /t %sleep_time% /nobreak > nul
-)
-
-REM If it failed all n_retries times, we can give up at last.
+@echo off
```
[the remark this answers is not in the document]
Comment (mozbhearsum) [reply lib/python/vendor/python-dateutil-2.6.0/ci_tools/retry.bat]: Yeah, this was accidental. I think it's just the line endings...

**Comment by mozbhearsum** [comment]
Here's a revised list of what's doing removed:
* All of `buildbot-helpers/`
* All of `buildfarm/breakpad` (but not the toplevel `breakpad`, which contain the still used minidump binaries)
* Most of `buildfarm/maintenance`. Most of this is buildbot stuff. The non-buildbot parts are: `archive-debug-builds.sh`, `count_and_reboot.py`, and `purge_builds.py`.
* `release-runner.py`/`release-runner.sh` (but not `release-runner3*`)
* All of `buildfarm/utils` except for `gittool.py`. Again, most of this is buildbo stuff with the exception of `retry.py` (replaced by redo everywhere that matters AFAICT), `mar.py` (replaced by mardor`), `hgtool.py` (replaced by robustcheckout).
* `clobberer`
* `lib/perl` (unused now that patcher-configs are dead)
* Much of lib/python, except the parts that I can find being used. The dying parts are: `build/checksums.py`, `build/misc.py`, `build/paths.py`, `build/upload.py`, `release/config.py`, `release/download.py`, `release/partials.py`, `release/sanity.py`, `release/update/patcher.py`, `slavealloc`, `util/fabric`, `util/post_file.py`, `util/transfer.py`, `util/tuxedo.py`.
* `release/add-promptWaitTime.py`, `release/firefox-tuxedo.ini`, `release-get-update-xml.sh`, `release/patcher-configs`, `release/test-mar-url.sh`, `release-tuxedo-add.py`
* Unused configs in `release/updates` (which is all of them)
* Lots of old stuff in scripts: peridioc update stuff (moved in-tree), old balrog submission scripts, old update verify config script, fuzzing scripts, spidermonkey build scripts (moved in tree), various other unused release automation scripts (all either deprecated or replaced with mozharness)
* All of `stage` -- which binaries that we used to use on the ancient stage.mozilla.org server (when it was ssh-able)

**Comment by mozbhearsum** [comment]
Pushed as https://hg.mozilla.org/build/tools/rev/d5289f1ceaaa

**Pull request closed without merging**